src/merge.ts

```diff
@@ -20,12 +20,12 @@ export const mergeResponses = mergeLists<IHttpOperationResponse>(
 );
 
 const mergeServers = mergeLists<IServer>(
-  (s1, s2) => s1.url === s2.url,
+  (s1, s2) => s1.url.toLowerCase() === s2.url.toLowerCase(),
```

Comment:
This is interesting, I do not know whether we should do this or not. urls are not case insensitive and some server do discriminate based on them. @marbemac @lottamus

Reply:
Actually @philsturgeon you could give your idea on this too.

Reply:
I don’t know how many people are doing this in the wild but yeah technically they could. Let’s leave this out and see if it’s reported, either by reports we can run or by users.

Reply:
I just removed it.

I believe we should have it like that: case-insensitive host and case-sensitive path. WDYT?

Reply:
I agree, HOSTS should be case insensitive. It this URL a full one (comprehensive of the host)?

Reply:
Well, that is even funnier. See the typings: https://github.com/stoplightio/types/blob/master/src/servers.ts#L8

We can have variables inside the server URL. I'll check this out, but it seems that we need to parse the URL in order to understand what is case-(in)?sensitive.

1) host name is case-insensitive
2) URL is case-sensitive
3) variable name: I can't find in OAS spec whether this is case sensitive or not, but I would assume it is

Now the funny part: a variable inside hostname makes it partially case-sensitive.

I guess we need to address it now or create an issue to address it later. Actually I'm also thinking about `http-spec` being responsible for server.url sanitization while transforming from different spec formats.

Reply:
Yeah let's make a separate issue for this, we can get this over the finish line. Feel free to open an issue.

Reply:
Here is the task: https://github.com/stoplightio/http-spec/issues/84